rclcpp/include/rclcpp/timer.hpp

```diff
@@ -33,7 +33,9 @@ namespace rclcpp
 namespace timer
 {
 
-using CallbackType = std::function<void()>;
+/// The timer callback signature returns a boolean.
+// If this callback returns true, then the timer should cancel.
```

Comment:
This surprised me when looking at the changes to the demos and tests. I was originally thinking that returning false would cancel the timer. But I don't have a good reason for that. To me neither "return false to indicate the timer should not continue firing" nor "return true to cancel the timer" are particularly intuitive. Perhaps we should return an typed enum so it's either `return timer::continue;` or `return timer::cancel;`. Not sure it's worth the effort, but see a bare `return false;` at the end of the timer callbacks left me guessing what that meant.